supabase/migrations/20241115111117_add-slug-in-topic.sql

```diff
@@ -0,0 +1,19 @@
+ALTER TABLE topics
+    ADD COLUMN slug text;
+
+UPDATE
+    topics
+SET
+    slug = REGEXP_REPLACE(REPLACE((
+            SELECT
+                t2.name
+            FROM topics t2
+            WHERE
+                topics.id = t2.id), ' ', '-'), '[^A-Za-z0-9-]', '', 'g');
+
+ALTER TABLE topics
+    ALTER COLUMN slug SET NOT NULL;
+
+ALTER TABLE topics
+    ADD CONSTRAINT slug_unique UNIQUE (slug);
+
```

Comment:
Consider adding an additional constraint that uses a function to validate if the value has a valid pattern.
Here's an example:
```sql
CREATE OR REPLACE FUNCTION public.valid_username(username text)
 RETURNS boolean
 LANGUAGE plpgsql
 SECURITY DEFINER
AS $function$
  BEGIN
    -- Only allow alphanumeric characters, underscores and dashes
    -- in usernames. This should also allow uuids
    RETURN username ~ '^[[:alnum:]_-]+$';
  END;
$function$
;
alter table "public"."user_info" add constraint valid_username check (valid_username(username));
```